src/Data/Vec/Properties.agda

```diff
@@ -1002,6 +999,29 @@ zipWith-replicate₂ _⊕_ []       y = refl
 zipWith-replicate₂ _⊕_ (x ∷ xs) y =
   cong (x ⊕ y ∷_) (zipWith-replicate₂ _⊕_ xs y)
 
+------------------------------------------------------------------------
+-- iterate
+
+length-iterate : ∀ n f (x : A) → length (iterate f x {n}) ≡ n
+length-iterate zero    f x = refl
+length-iterate (suc n) f x = cong suc refl
+
+iterate-id : ∀ n (x : A) → iterate id x {n} ≡ replicate x
+iterate-id zero x    = refl
+iterate-id (suc n) x = cong (_ ∷_) (iterate-id n (id x))
+
+take-iterate : ∀ n f (x : A) → take n (iterate f x {n + m}) ≡ iterate f x
+take-iterate zero    f x = refl
+take-iterate (suc n) f x = cong (_ ∷_) (take-iterate n f (f x))
+
+drop-iterate : ∀ n f (x : A) → drop n (iterate f x) ≡ []
```

Comment:
Is there an 'easy' analogue of `List.Properties.drop-all` which you can derive this from?